projects/hip-tests/catch/unit/module/hipModuleGetLoadingMode.cc

```diff
@@ -0,0 +1,168 @@
+LIABILITY, WHETHER INN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR INN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+#include <hip_test_common.hh>
+#include <hip_test_defgroups.hh>
+constexpr int LEN = 64;
+constexpr auto SIZE = (LEN << 2);
+constexpr auto codeObjFile1 = "vcpy_kernel.code";
+constexpr auto kernel_name1 = "hello_world";
+constexpr auto codeObjFile2 = "copyKernel.code";
+constexpr auto kernel_name2 = "copy_ker";
+/**
+ * @addtogroup hipModuleGetLoadingMode hipModuleGetLoadingMode
+ * @{
+ * @ingroup ModuleTest
+ * `hhipError_t hipModuleGetLoadingMode(hipModuleLoadingMode_t* mode)` -
+ * Function gets the current module load mode
+ */
+
+/**
+ * Test Description
+ * ------------------------
+ * - Test case verifies the positive case of hipModuleGetLoadingMode API.
+ * - Verifies the default mode. Default mode is Lazy.
+ * Test source
+ * ------------------------
+ * - catch/unit/module/hipModuleGetLoadingMode.cc
+ * Test requirements
+ * ------------------------
+ * - HIP_VERSION >= 7.2
+ */
+TEST_CASE("Unit_hipModuleGetLoadingMode_DefaultModeCheck") {
+  hipModuleLoadingMode_t mode;
+  HIP_CHECK(hipModuleGetLoadingMode(&mode));
+  REQUIRE(mode == HIP_MODULE_LAZY_LOADING);
+}
+/**
+ * Test Description
+ * ------------------------
+ * - Test case verifies the positive case of hipModuleGetLoadingMode API.
+ * - Verifies the Eager mode after setting the env var HIP_MODULE_LOADING.
+ * Test source
+ * ------------------------
+ * - catch/unit/module/hipModuleGetLoadingMode.cc
+ * Test requirements
+ * ------------------------
+ * - HIP_VERSION >= 7.2
+ */
+TEST_CASE("Unit_hipModuleGetLoadingMode_EagerModeCheck") {
+  hipModuleLoadingMode_t mode;
+  if (setenv("HIP_MODULE_LOADING", "EAGER", 1) != 0 ) {
+    HIP_CHECK(hipModuleGetLoadingMode(&mode));
+    REQUIRE(mode == HIP_MODULE_LAZY_LOADING);  // Default mode is lazy mode
+  } else {
+    HIP_CHECK(hipModuleGetLoadingMode(&mode));
+    REQUIRE(mode == HIP_MODULE_EAGER_LOADING);  // if env var HIP_MODULE_LOADING = EAGER
+  }
+  unsetenv("HIP_MODULE_LOADING");
+}
+/**
+ * Test Description
+ * ------------------------
+ * - Test case verifies the positive case with kernel Launch.
+ * Test source
+ * ------------------------
+ * - catch/unit/module/hipModuleGetLoadingMode.cc
+ * Test requirements
+ * ------------------------
+ * - HIP_VERSION >= 7.2
+ */
+TEST_CASE("Unit_hipModuleGetLoadingMode_modeBasedKernel") {
+  float *Ad, *Bd;
+  std::vector<float> A(LEN, 1.0f), B(LEN, 0.0f);
+
+  HIP_CHECK(hipMalloc(&Ad, SIZE));
+  HIP_CHECK(hipMalloc(&Bd, SIZE));
+  HIP_CHECK(hipMemcpy(Ad, A.data(), SIZE, hipMemcpyHostToDevice));
+  HIP_CHECK(hipMemcpy(Bd, B.data(), SIZE, hipMemcpyHostToDevice));
+
+  hipModule_t Module;
+  hipFunction_t Function = nullptr;
+  hipModuleLoadingMode_t mode;
+  HIP_CHECK(hipModuleGetLoadingMode(&mode));
+  if (mode == HIP_MODULE_EAGER_LOADING) {
+    HIP_CHECK(hipModuleLoad(&Module, codeObjFile1));
```

Comment:
hipModuleLoad in EAGER mode should take longer time than hipModuleLoad in LAZY load. 
Its mainly performance related testing. 
getting of variables and functions is deferred to later point till we getFunction and getVariable for Lazy.